server/controllers/getProduct.js

```diff
@@ -0,0 +1,10 @@
+
+const getProduct = (req,res)=>{
+    const {productId}=req.params;
+    getSingleProduct(productId)
+    .then((data) => res.json(data.rows))
```

Comment:
I think you need to send the first element of the rows array, check it and destructure it
```suggestion
    .then(({ rows }) => res.json(rows[0]))
```